### PR TITLE
added saving and loading capability for scenarios

### PR DESCRIPTION
**Pull request by yangdanny97**
saves to resources/public/scenarios as a .txt file with the name specified
load scenarios with the specified name into scenario viewer
add scenarios to world, which resets the world and removes all bots before loading scenario objects

### CODE REVIEW

File: cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java [hunk 1]
```diff
@@ -129,11 +140,58 @@ public static void main(String[] args) {
                         (float)position[1], size, angle);
                 simvs.importPhysicalObject(po);
             }
-            /* storing json objects into actual variables */
+
+
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java]: can you reduce the whitespace here? just do one newline

File: cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java [hunk 2]
```diff
@@ -129,11 +140,58 @@ public static void main(String[] args) {
...
 
             return addInfo;
         });
 
+        //saves a scenario as a file
+        post("/saveScenario", (req,res) -> {
+
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java]: remove newline here

File: cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java [hunk 3]
```diff
@@ -129,11 +140,58 @@ public static void main(String[] args) {
...
+            String body = req.body();
+            JsonObject scenario = jp.parse(body).getAsJsonObject();
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java]: can you refactor jp to be jsonParser? (shift+f6 does this in intelliJ)

File: cs-minibot-platform-src/src/main/resources/public/gui/index.html [hunk 4]
```diff
@@ -107,9 +107,13 @@
 		<button id="addBot">add bot</button>
 		<button id = "updateLocs"> Run simulator </button>
 		<br><hr><br>
-		Scenario:
-		<textarea rows="4" cols="50" name="scenario" id="scenario">[{"type": "scenario_object", "position":"[5,5]", "angle":45, "size":1},{"type": "scenario_object", "position":"[3,3]", "angle":45, "size": 1}]</textarea>
+		Scenario name:<input type="text" id="scenarioname"
+							 value="scenario1"><br>
+		Scenario viewer: <textarea rows="4" cols="50" name="scenario"
+					   id="scenario">[{"type":	"scenario_object", "position":"[5,5]", "angle":45, "size":1},{"type": "scenario_object", "position":"[3,3]", "angle":45, "size": 1}]</textarea>
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/resources/public/gui/index.html]: is it possible to load a bot in the scenario? not clear from this json. if so- add a bot to this scenario so it is easy for people to pickup

File: cs-minibot-platform-src/src/main/resources/public/gui/index.html [hunk 5]
```diff
@@ -107,9 +107,13 @@
...
 		<button id="addScenario">add scenario</button>
+		<button id="loadScenario">load scenario</button>
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/resources/public/gui/index.html]: maybe rename these buttons to be extra clear to everyone else on the team- from what you showed me, I would see add scenario as load scenario from text box, load scenario as fetch scenario to text box, and save scenario as save scenario from textbox

File: cs-minibot-platform-src/src/main/resources/public/js/interface.js [hunk 6]
```diff
@@ -182,11 +182,54 @@ $('#addScenario').click(function() {
...
+     console.log("saving a scenario")
+     var scenario = $("#scenario").val();
+     var filename = $("#scenarioname").val();
+
+     $.ajax({
+         method: "POST",
+         url: '/saveScenario',
+         dataType: 'text',
+         data: JSON.stringify({'scenario': scenario.toString(),
+         'name':filename.toString()}),
```
Comment (TrevorEdwards) [inline cs-minibot-platform-src/src/main/resources/public/js/interface.js]: don't use a string key here- inconsistent with everything else

**Comment by TrevorEdwards** [comment]
@yangdanny97 needs merge from develop

File: cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java [hunk 7]
```diff
@@ -96,48 +101,151 @@ public static void main(String[] args) {
...
+
+            FileWriter writer = new FileWriter(file, false);
+            BufferedWriter bwriter = new BufferedWriter(writer);
+            bwriter.write(scenarioBody);
+            bwriter.close();
+            out.close();
+            return fileName;
+        });
+
+        /*loads a scenario with the specified file name to the simulator
```
Comment (alan20854) [inline cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java]: hi danny which part of the code does this floating comment belong to? Trevor would like it to be more obvious

**Pull request merged**